addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -73,6 +70,59 @@ publicWidget.registry.anchorSlide.include({
     },
 });
 
+baseAnimatedHeader.include({
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     * @private
+     */
+    _updateMainPaddingTop: function () {
+        this._super(...arguments);
+        this._updateTableOfcontentNavbarPosition();
+    },
+    /**
+     * @private
+     */
+    _updateTableOfcontentNavbarPosition: function () {
+        const $navbar = this.$el.parent().find('.s_table_of_content_navbar_sticky');
+        if ($navbar.length) {
```

Comment:
How does this work when there is two tables of content on a page? I think there is a clean solution to handle that which will not require an `include`.

Instead of "include the menu class so that on each padding-top update, it searches for all table of content navbar which are sticking then update them", you can:
- In the menu widgets directly: call a series of registered callbacks
- Register a callback when necessary = when you have table of content with a top sticky navbar

That way the code is still well separated, efficient and without include.
menu:
```javascript
extraMenuUpdateCallbacks = [];
BaseAnimatedHeader = ... {
    _updateMainPaddingTop: function () {
         const headerHeight = ...;
         paddingTop = ...;
         for (const callback of extraMenuUpdateCallbacks) {
              callback(headerHeight);
         }
    },
});

....

return {
     extraMenuUpdateCallbacks: extraMenuUpdateCallbacks,
};
```

table of content:
```javascript
const {extraMenuUpdateCallbacks} = require(...);
TableOfContent = ... {
    start: function () {
         if (stickyTopNav) {
                extraMenuUpdateCallbacks.push(this._updateMachin.bind(this));
         }
    },
   _updateMachin: function (headerHeight) { ... }
});
```